src/game/client/gameclient.cpp

```diff
@@ -14,15 +14,503 @@
+    }
+
+    if (g_theKeyboard != nullptr) {
+        delete g_theKeyboard;
+        g_theKeyboard = nullptr;
+    }
+
+    if (g_theDisplayStringManager != nullptr) {
+        delete g_theDisplayStringManager;
+        g_theDisplayStringManager = nullptr;
+    }
+
+    if (g_theEva != nullptr) {
+        delete g_theEva;
+        g_theEva = nullptr;
+    }
+
+    if (g_theSnowManager != nullptr) {
+        delete g_theSnowManager;
+        g_theSnowManager = nullptr;
+    }
+}
+
+void GameClient::Init()
+{
+    Set_Frame_Rate(33.333332f);
+    INI ini;
+    ini.Load("Data\\INI\\DrawGroupInfo.ini", INI_LOAD_OVERWRITE, nullptr);
+
+    if (g_theGlobalLanguage != nullptr && g_theGlobalLanguage->Draw_Group_Info_Font().Name().Is_Not_Empty()) {
+        g_theDrawGroupInfo->m_fontName = g_theGlobalLanguage->Draw_Group_Info_Font().Name();
+        g_theDrawGroupInfo->m_fontSize = g_theGlobalLanguage->Draw_Group_Info_Font().Point_Size();
+        g_theDrawGroupInfo->m_fontIsBold = g_theGlobalLanguage->Draw_Group_Info_Font().Bold();
+    }
+
+    g_theDisplayStringManager = Create_DisplayStringManager();
+
+    if (g_theDisplayStringManager != nullptr) {
+        g_theDisplayStringManager->Init();
+        g_theDisplayStringManager->Set_Name("TheDisplayStringManager");
+    }
+
+    g_theKeyboard = Create_Keyboard();
+    g_theKeyboard->Init();
+    g_theKeyboard->Set_Name("TheKeyboard");
+
+    g_theMappedImageCollection = new ImageCollection();
+    g_theMappedImageCollection->Load(512);
+
+    g_theAnim2DCollection = new Anim2DCollection();
+    g_theAnim2DCollection->Init();
+    g_theAnim2DCollection->Set_Name("TheAnim2DCollection");
+
+    if (g_theMessageStream != nullptr) {
+        m_translators[m_translatorCount++] = g_theMessageStream->Attach_Translator(new WindowTranslator(), 10);
+        m_translators[m_translatorCount++] = g_theMessageStream->Attach_Translator(new MetaEventTranslator(), 20);
+        m_translators[m_translatorCount++] = g_theMessageStream->Attach_Translator(new HotKeyTranslator(), 25);
+        m_translators[m_translatorCount++] = g_theMessageStream->Attach_Translator(new PlaceEventTranslator(), 30);
+        m_translators[m_translatorCount++] = g_theMessageStream->Attach_Translator(new GUICommandTranslator(), 40);
+        m_translators[m_translatorCount++] = g_theMessageStream->Attach_Translator(new SelectionTranslator(), 50);
+        m_translators[m_translatorCount++] = g_theMessageStream->Attach_Translator(new LookAtTranslator(), 60);
+        m_translators[m_translatorCount] = g_theMessageStream->Attach_Translator(new CommandTranslator(), 70);
+        m_commandTranslator =
+            static_cast<CommandTranslator *>(g_theMessageStream->Find_Translator(m_translators[m_translatorCount]));
+        m_translatorCount++;
+        m_translators[m_translatorCount] = g_theMessageStream->Attach_Translator(new HintSpyTranslator(), 100);
+        m_translators[m_translatorCount] =
```

Comment:
m_translatorCount++

I think, because:
```c++
    for (unsigned int i = 0; i < m_translatorCount; i++) {
        g_theMessageStream->Remove_Translator(m_translators[i]);
    }
```

Reply:
fixed.